.github/workflows/publish-on-stackblitz.yml

```diff
@@ -24,9 +24,10 @@ jobs:
         uses: actions/cache@v4
         with:
           path: .turbo
-          key: ${{ runner.os }}-turbo-node-20
+          key: turbo-${{ runner.os }}-node-20-branch-${{ github.head_ref }}
           restore-keys: |
-            ${{ runner.os }}-turbo-node-
+            turbo-${{ runner.os }}-node-20-branch-
+            turbo-${{ runner.os }}-node-
```

Comment:
Nice work! I can see the build cache being hit by all these other test workflows. I wasn't understanding how that part works. I assumed that github ran these workflows in parallel? Is there only redundancy on the first push to the branch?

Reply:
Yes! It tries to restore the cache from those sources:

1. the exact same branch: turbo-${{ runner.os }}-node-20-branch-${{ github.head_ref }}
2. any other branch: turbo-${{ runner.os }}-node-20-branch-*
3. or just the same Node version: turbo-${{ runner.os }}-node-*

The first commit for a PR should restore the cache from any other branch, for the following commits it’s taking the cache from the exact same branch.

> **restore-keys:** Optional A string containing alternative restore keys, with each restore key placed on a new line. If no cache hit occurs for key, these restore keys are used sequentially in the order provided to find and restore a cache. For example:

Source: https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#using-the-cache-action

Reply:
Thanks for the clarification. That makes sense now 😄